### ISSUE TITLE
Embassy STM32 examples not providing timestamps on defmt logs

### ISSUE THREAD

**Issue by henryshunt** [embassy-rs/embassy#3569]
When I run the STM32 examples, I don't see timestamps on the `defmt` logs as the getting started docs show, despite the examples apparently having the necessary configuration to provide timestamps. I've also been unable to get timestamps when creating my own Embassy projects.

If I change the `cargo run` command to explicitly display timestamps in `probe-rs`, I get timestamps (presumably from the host side rather than the target) and the following message is output:

```
WARN defmt_decoder::log::format: logger format contains timestamp but no timestamp implementation was provided; consider removing the timestamp (`{t}` or `{T}`) from the logger format or provide a `defmt::timestamp!` implementation
```

This seems to imply the target isn't actually providing timestamps to the host.

Does anyone have any idea how to solve this and get timestamps from the target? My `probe-rs` version is v0.24.0.

**Comment by henryshunt** [comment]
Digging into this further, I've confirmed the target is in fact sending timestamps to the host. If I remove the `defmt-timestamp-uptime-*` feature from `embassy-time`, instead of `probe-rs` displaying a timestamp it displays `<time>`. This implies the timestamps I'm getting when I explicitly add `{t}` to `--log-format` are from the target and not the host as I had presumed.

Additionally, inspecting the raw RTT data from the target shows extra bytes included when the `defmt-timestamp-uptime-*` feature is enabled vs when not.

So it appears this is an issue with something on the host side not properly detecting that the target is providing timestamps.
